purchases/src/main/kotlin/com/revenuecat/purchases/PostReceiptHelper.kt

```diff
@@ -0,0 +1,143 @@
+package com.revenuecat.purchases
+
+import com.revenuecat.purchases.common.AppConfig
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.BillingAbstract
+import com.revenuecat.purchases.common.PostReceiptDataErrorCallback
+import com.revenuecat.purchases.common.PostReceiptDataSuccessCallback
+import com.revenuecat.purchases.common.ReceiptInfo
+import com.revenuecat.purchases.common.caching.DeviceCache
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.models.StoreTransaction
+import com.revenuecat.purchases.subscriberattributes.SubscriberAttributesManager
+import com.revenuecat.purchases.subscriberattributes.getAttributeErrors
+import com.revenuecat.purchases.subscriberattributes.toBackendMap
+
+@Suppress("LongParameterList")
+internal class PostReceiptHelper(
+    private val appConfig: AppConfig,
+    private val backend: Backend,
+    private val billing: BillingAbstract,
+    private val customerInfoHelper: CustomerInfoHelper,
+    private val deviceCache: DeviceCache,
+    private val subscriberAttributesManager: SubscriberAttributesManager
+) {
+    private val finishTransactions: Boolean
+        get() = appConfig.finishTransactions
+
+    /**
+     * This method will post a token and receiptInfo to the backend without consuming any purchases.
+     * It will store that the token was sent to the backend so it doesn't send it again.
+     */
+    fun postTokenAndReceiptInfoToBackend(
```

Comment:
feels like the "not consuming" part is important enough that it should be an explicit part of the signature. It has major implications after all. 

Maybe `postTokenWithoutConsuming` or something? The `toBackend` is probably one part we can do without since we always post to backend, there won't be any confusion there.

Reply:
👍 Agreed! Much better name